web/src/test/java/se/inera/intyg/rehabstod/service/sjukfall/UnansweredCommunicationDecoratorServiceTest.java

```diff
@@ -0,0 +1,157 @@
+    private static final List<SjukfallEnhet> SICK_LEAVES = List.of(
+            SICK_LEAVE_ONE_CERTIFICATE,
+            SICK_LEAVE_NO_CERTIFICATE_WITH_COMMUNICATION,
+            SICK_LEAVE_SEVERAL_CERTIFICATES
+    );
+    private static final UnansweredQAs QA_0 = new UnansweredQAs(0, 0);
+    private static final UnansweredQAs QA_1 = new UnansweredQAs(5, 3);
+    private static final UnansweredQAs QA_2 = new UnansweredQAs(2, 3);
+    private static final UnansweredQAs QA_3 = new UnansweredQAs(10, 20);
+
+    @BeforeEach
+    void setup() {
+        SICK_LEAVE_SEVERAL_CERTIFICATES.setIntygLista(Arrays.asList(CERTIFICATE_ID_0, CERTIFICATE_ID_1, CERTIFICATE_ID_2));
+        SICK_LEAVE_SEVERAL_CERTIFICATES.setPatient(new Patient(PATIENT_ID_0, "Name"));
+        SICK_LEAVE_ONE_CERTIFICATE.setIntygLista(List.of(CERTIFICATE_ID_3));
+        SICK_LEAVE_ONE_CERTIFICATE.setPatient(new Patient(PATIENT_ID_1, "Name"));
+        SICK_LEAVE_NO_CERTIFICATE_WITH_COMMUNICATION.setIntygLista(List.of("NO_COMMUNICATION"));
+        SICK_LEAVE_NO_CERTIFICATE_WITH_COMMUNICATION.setPatient(new Patient(PATIENT_ID_2, "Name"));
+
+        final var unansweredQAsMap = new HashMap<String, UnansweredQAs>();
+        unansweredQAsMap.put(CERTIFICATE_ID_0, QA_0);
+        unansweredQAsMap.put(CERTIFICATE_ID_1, QA_1);
+        unansweredQAsMap.put(CERTIFICATE_ID_2, QA_2);
+        unansweredQAsMap.put(CERTIFICATE_ID_3, QA_3);
+        response.setUnansweredQAsMap(unansweredQAsMap);
+
+        when(wcRestIntegrationService.getUnansweredCommunicationForPatients(anyList())).thenReturn(response);
+    }
+
+    @Nested
+    class TestItRequest {
+
+        @Test
+        void shouldCallRestServiceWithPatientIds() {
+            final var captor = ArgumentCaptor.forClass(List.class);
+
+            unansweredCommunicationDecoratorService.decorate(SICK_LEAVES);
+
+            verify(wcRestIntegrationService).getUnansweredCommunicationForPatients(captor.capture());
+            assertEquals(SICK_LEAVES.size(), captor.getValue().size());
+            assertTrue(captor.getValue().contains(PATIENT_ID_0));
+            assertTrue(captor.getValue().contains(PATIENT_ID_1));
+            assertTrue(captor.getValue().contains(PATIENT_ID_2));
+        }
+    }
+
+    @Nested
+    class TestResponse {
+        @Test
+        void shouldSetErrorAsTrueIfResponseIsSuccess() {
+            response.setUnansweredCommunicationError(true);
+
+            final var hasDecorated = unansweredCommunicationDecoratorService.decorate(SICK_LEAVES);
+
+            assertFalse(hasDecorated);
+        }
+
+        @Test
+        void shouldSetErrorAsFalseIfResponseIsFail() {
+            response.setUnansweredCommunicationError(false);
+
+            final var hasDecorated = unansweredCommunicationDecoratorService.decorate(SICK_LEAVES);
+
+            assertTrue(hasDecorated);
+        }
+    }
+
+    @Nested
+    class TestDecoration {
```

Comment:
Would it make sense to split these tests into two separeted ``` @Nested class Unanswered-Other/Kompl``` ?

That way we assure that the tests are testing one thing and we know if we got issues with the logic related to other or komplettering if something would break in the future.